tiledb/ml/models/base.py

```diff
@@ -73,38 +104,37 @@ def preview(self, **kwargs):
         PyTorch etc.
         """
 
-    def set_file_properties(
-        self, framework: str, framework_version: str, preview: str
-    ) -> dict:
+    def set_file_properties(self):
         """
-        Abstract method that returns model array's file properties.
-        :param framework: str. Machine learning framework.
-        :param framework_version: str. Machine learning framework's version.
-        :param preview: str. Machine learning model's preview.
-        :return: dict. Dictionary with model's file properties
+        Method that sets model array's file properties.
         """
-        return {
-            self.file_properties_obj.ML_FRAMEWORK.value: framework,
-            self.file_properties_obj.STAGE.value: "STAGING",
-            self.file_properties_obj.PYTHON_VERSION.value: platform.python_version(),
-            self.file_properties_obj.FRAMEWORK_VERSION.value: framework_version,
-            self.file_properties_obj.MODEL_PREVIEW.value: preview,
+        self._file_properties = {
+            ModelFileProperties.ML_FRAMEWORK.value: self.framework,
+            ModelFileProperties.ML_FRAMEWORK_VERSION.value: self.framework_version,
+            ModelFileProperties.STAGE.value: "STAGING",
+            ModelFileProperties.PYTHON_VERSION.value: platform.python_version(),
+            ModelFileProperties.PREVIEW.value: self.preview(),
         }
 
-    @staticmethod
-    def update_model_metadata(array: tiledb.Array, meta: dict):
-        for key, value in meta.items():
-            array.meta[key] = json.dumps(value).encode("utf8")
-
-
-@unique
-class ModelFileProperties(Enum):
-    """
-    Enum Class that contains all model array file properties.
-    """
+    def update_model_metadata(self, array: tiledb.Array, meta: dict):
+        """
+        This method updates the metadata in a TileDB model array. File properties also go in the metadata section.
+        :param array: tiledb.Array. A TileDB model array.
+        :param meta: dict. A dictionary with the <key, value> pairs that will be inserted in array's metadata.
+        """
+        # Raise ValueError in case users provide metadata with the same keys as file properties.
+        if meta:
+            if meta.keys() & self._file_properties.keys():
+                raise ValueError(
+                    "Please avoid using file property key names as metadata keys!"
+                )
+            else:
+                for key, value in {**meta, **self._file_properties}.items():
+                    array.meta[key] = json.dumps(value).encode("utf8")
+        else:
+            # In case meta is None, insert only file properties in the metadata.
```

Comment:
`meta` is annotated as `dict` so it shouldn't be None. Even if changes to `Optional[dict]` (or more generally `Optional[Mapping]`), you can change None to empty dict. In any case the if/else on `meta` can be removed.